### PR TITLE
[stable/postgresql] Add service account support

### LINKED ISSUE #13748: [stable/postgresql] Service Account support
**Is your feature request related to a problem? Please describe.**
Kubernetes service accounts provide an identity for processes running in a Pod. When combining Kubernetes with an Istio service mesh, these identities can be used to enforce authorization policies. Without a defined service account, deployments fall back to the default service account of the namespace the deployment resides in. This makes it impossible to identify a particular service and thus enforce security policies.

**Describe the solution you'd like**
Utilize service accounts when deploying Postgres. The chart should automatically create a service account and/or use an existing service account. The name of an existing service account should be configurable.

**Describe alternatives you've considered**
The alternative is not using the Helm chart or not utilizing Istio's security features.

### PR DESCRIPTION
**Pull request by denniseffing**
#### What this PR does / why we need it:
This PR adds Kubernetes service account support. Service account utilization is disabled by default but can be enabled by setting `serviceAccount.enabled=true`. If enabled, a new service account is created by default when installing the chart. Optionally, an existing service account can be used by setting `serviceAccount.name` (Note: Setting `serviceAccount.name` disables automatic service account creation.)

#### Which issue this PR fixes
Fixes #13748 

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

File: stable/postgresql/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: postgresql
-version: 4.0.3
+version: 4.1.3
```
Comment (juan131) [inline stable/postgresql/Chart.yaml]: The version should be `4.1.0`

File: stable/postgresql/values.yaml [hunk 2]
```diff
@@ -65,6 +65,12 @@ securityContext:
   fsGroup: 1001
   runAsUser: 1001
 
+## Pod Service Account
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
+serviceAccount:
+  enabled: false
+  # name:
```
Comment (juan131) [inline stable/postgresql/values.yaml:73]: I'd add more information about the implications on setting this parameter `serviceAccount.name`.

Please update the `values-production.yaml` with these changes too

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for your contribution!! Please take a look to my comments when you have a chance.

File: stable/postgresql/Chart.yaml (hunk 1 above)
Comment (denniseffing) [reply stable/postgresql/Chart.yaml]: Whoops, good catch! Done.

File: stable/postgresql/values.yaml (hunk 2 above)
Comment (denniseffing) [reply stable/postgresql/values.yaml:73]: Thanks, also done!

**Comment by juan131** [comment]
Hi @denniseffing 

I'm okay with the changes but I'm afraid there was a release of PostgreSQL to update it to `10.8.x` series and you'll need to rebase your PR from master and update the version again in the Chart.yaml.

**Comment by denniseffing** [comment]
Hi @juan131 

I rebased the branch and updated the version again. Thanks for your reviewing time!

**Pull request merged**